x/sync/firewood/syncer.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package firewood
+
+import (
+	"bytes"
+	"context"
+	"errors"
+
+	"github.com/ava-labs/firewood-go-ethhash/ffi"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/maybe"
+
+	xsync "github.com/ava-labs/avalanchego/x/sync"
+)
+
+var (
+	_ xsync.DB[*RangeProof, struct{}] = (*database)(nil)
+
+	defaultSimultaneousWorkLimit = 8
+)
+
+// database wraps a Firewood FFI database to implement the xsync.DB interface.
+type database struct {
+	db *ffi.Database
+}
+
+type Config struct {
+	SimultaneousWorkLimit int                   // optional, defaults to 8
+	Log                   logging.Logger        // optional
+	StateSyncNodes        []ids.NodeID          // optional
+	Registerer            prometheus.Registerer // optional
+}
+
+func NewSyncer(config Config, db *ffi.Database, targetRoot ids.ID, rangeProofClient *p2p.Client, changeProofClient *p2p.Client) (*xsync.Syncer[*RangeProof, struct{}], error) {
+	r := config.Registerer
+	if r == nil {
+		r = prometheus.NewRegistry()
+	}
+	logger := config.Log
+	if logger == nil {
+		logger = logging.NoLog{}
+	}
+	workLimit := config.SimultaneousWorkLimit
+	if workLimit <= 0 {
+		workLimit = defaultSimultaneousWorkLimit
+	}
```

Comment:
This is deviating a bit from the zero values -> override pattern configs use, which creates a misleading api since if the caller intentionally specifies a negative value we actually override it (we're mixing responsibilities of default setting and sanitization). I recommend throwing an error if this is negative (I think x/sync needs a regression test on this).

```suggestion
	if workLimit == 0 {
		workLimit = defaultSimultaneousWorkLimit
	}
```